tools/distpartitioning/globalids.py

```diff
@@ -29,41 +29,18 @@ def get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data):
         from other processes. 
     """
     #build a list of sizes (lengths of lists)
-    sizes = [len(x) for x in global_nids_ranks]
-
-    #compute total_nodes whose mappings should be resolved, between orig-node-id <-> global-id
-    total_nodes = np.sum(sizes)
-    if (total_nodes == 0):
-        print('Rank: ', rank, ' -- All mappings are present locally... No need for to send any info.')
-        return None
-
-    #determine the no. of global_node_ids to send and receive and perform alltoall
-    send_counts = list(torch.Tensor(sizes).type(dtype=torch.int64).chunk(world_size))
-    recv_counts = list(torch.zeros([world_size], dtype=torch.int64).chunk(world_size))
-    alltoall_cpu(rank, world_size, recv_counts, send_counts)
-
-    #allocate buffers to receive node-ids
-    recv_nodes = []
-    for i in recv_counts:
-        recv_nodes.append(torch.zeros(i.tolist(), dtype=torch.int64))
-
-    #form the outgoing message
-    send_nodes = []
-    for i in range(world_size):
-        send_nodes.append(torch.from_numpy(global_nids_ranks[i]).type(dtype=torch.int64))
-
-    #send-recieve messages
-    alltoallv_cpu(rank, world_size, recv_nodes, send_nodes)
-
-    # allocate buffers to receive global-ids
-    recv_shuffle_global_nids = []
-    for i in sizes: 
-        recv_shuffle_global_nids.append(torch.zeros((i), dtype=torch.int64))
+    sizes = [x.shape for x in global_nids_ranks]
```

Comment:
where is `sizes` used?

Reply:
Correct. This variable is not used anymore. Prior to alltoall_cpu_data, when alltoallv was used sizes was used to create appropriate buffers on the receiving side. This is irrelevant now. 

Removed 'sizes'. 

Please note that the logic of creating buffers for receiving messages is created inside the alltoall_cpu_data function call.